### ISSUE TITLE
S3 access denied when adding storage to existing project

### ISSUE THREAD

**Issue by antantonton** [aws-amplify/amplify-js#6302]
**Describe the bug**
When adding storage to an existing project with Auth and API enabled, any Storage.get, Storaget.put and Storage.list operation returns with error 403 (forbidden), see screenshot below.

**To Reproduce**
1. Existing project with Auth and API added
2. amplify add storage
3. amplify push
4. In main.ts, run Amplify.configure() with the aws-exports.js input given below
5. Run code snippet below

**Expected behavior**
Expected that following the start-up guide for Storage would result in a usable S3 bucket.

**Code Snippet**
```
    Storage.put('test.txt', 'Hello')
    .then (result => console.log(result))
    .catch(err => console.log(err));
```

**Screenshots**
![image](https://user-images.githubusercontent.com/37895272/87234197-0eabad00-c3cf-11ea-9226-d6220763a0d8.png)

**What is Configured?**
* Which steps did you follow via Amplify CLI when configuring your resources.
* Which resources do you have configured? Auth, API
  * If applicable, please provide your `aws-exports` file:
{
    "aws_project_region": "eu-west-1",
    "aws_cognito_identity_pool_id": "eu-west-1:xxx",
    "aws_cognito_region": "eu-west-1",
    "aws_user_pools_id": "eu-west-xxx",
    "aws_user_pools_web_client_id": "xxx",
    "oauth": {},
    "aws_appsync_graphqlEndpoint": "xxx",
    "aws_appsync_region": "eu-west-1",
    "aws_appsync_authenticationType": "AMAZON_COGNITO_USER_POOLS",
    "aws_user_files_s3_bucket": "xxx",
    "aws_user_files_s3_bucket_region": "eu-west-1"
}

**Additional notes**
I have tried multiple actions to remedy this without success:
- Adding complete S3 access to both auth and unAuth roles in IAM
- Giving the S3 bucket full public access
- Completely removing auth and storage from the project and then re-adding them
- Running API, PubSub, Auth and Storage .configure with aws-exports.js as input in main.ts
- Updated aws-amplify to the latest version
- Cognito user without any groups
- Cognito user with a group that has role set to the authRole

**Comment by antantonton** [comment]
Small update.

After banging my head at this for a couple of more hours, I finally found what seems to be the cause of my authentication problems. I had removed auth from the project and added it again to change from username to email login. During this procedure the authRole and unAuthRole **did not update the trusted relationship to reference the new identity-pool name**.
I updated these manually and finally got S3 access back.

I am now cleaning out my project from storage and auth and re-doing things to see if this fully resolves the problem. I will let you know if this is successful.

I still consider this a bug however, removeing and re-adding auth should update the auth- and unAuthRoles to allowed the newly created identity-pool. As of now, the old and removed identity-pool is still referenced in the roles after running 'amplify push'
